### DISCUSSION TITLE
Guild member nickname

### DISCUSSION THREAD

**Discussion by moosah-edge** [discordjs/discord.js#6915] [Q&A] [answered]
Hi all,

Apologies, might have posted in the wrong section.  I'm trying to get a user's server nickname and I've been wracking my brain on this for hours. 

I've tried a number of things, however the closest I can get is to see the value I want in the console when I print the following.  However I can't seem to grab the values in this promise object.

```
var guildMember = message.guild.members.fetch(message.author);
console.log(guildMember);
```

Prints this: 

> Promise {
>   GuildMember {
>     guild: Guild {
>       id: '.....',
>       name: '.....',
>       icon: '.....',
>       features: [],
>       commands: [GuildApplicationCommandManager],
>       members: [GuildMemberManager],
>       channels: [GuildChannelManager],
>       bans: [GuildBanManager],
>       roles: [RoleManager],
>       presences: PresenceManager {},
>       voiceStates: [VoiceStateManager],
>       stageInstances: [StageInstanceManager],
>       invites: [GuildInviteManager],
>       deleted: false,
>       available: true,
>       shardId: 0,
>       splash: null,
>       banner: null,
>       description: null,
>       verificationLevel: 'NONE',
>       vanityURLCode: null,
>       nsfwLevel: 'DEFAULT',
>       discoverySplash: null,
>       memberCount: 401,
>       large: true,
>       applicationId: null,
>       afkTimeout: 3600,
>       afkChannelId: '.....',
>       systemChannelId: '.....',
>       premiumTier: 'NONE',
>       premiumSubscriptionCount: 0,
>       explicitContentFilter: 'DISABLED',
>       mfaLevel: 'NONE',
>       joinedTimestamp: 1634274571032,
>       defaultMessageNotifications: 'ALL_MESSAGES',
>       systemChannelFlags: [SystemChannelFlags],
>       maximumMembers: 250000,
>       maximumPresences: null,
>       approximateMemberCount: null,
>       approximatePresenceCount: null,
>       vanityURLUses: null,
>       rulesChannelId: null,
>       publicUpdatesChannelId: null,
>       preferredLocale: 'en-US',
>       ownerId: '.....',
>       emojis: [GuildEmojiManager],
>       stickers: [GuildStickerManager]
>     },
>     joinedTimestamp: 1515668511934,
>     premiumSinceTimestamp: null,
>     deleted: false,
>     nickname: 'TestNickname',
>     pending: false,
>     _roles: [ '.....', '.....' ],
>     user: User {
>       id: '*****',
>       bot: false,
>       system: false,
>       flags: [UserFlags],
>       username: '.....',
>       discriminator: '4058',
>       avatar: '.....',
>       banner: undefined,
>       accentColor: undefined
>     },
>     avatar: null
>   }
> }

Can someone please help me understand how to access these values?  All I'm trying to do is assign the value of nickname to a string variable.

I assumed it should be as simple as guildMember.nickname or something similar, but nothing I've tried works...

**Comment by ImRodry** [comment, marked as answer]
It is as simple as guildMember.nickname, you just complicated it way more than you had to. Since you’re using fetch you have to await your promise to then access the nickname. If the message is sent on a guild you can also just use message.member.nickname

**Reply by moosah-edge** [reply]
Holy god thank you.  Went down the bloody rabbit hole on this one.
